src/bosh-director/lib/bosh/director/deployment_plan/stages/create_network.rb

```diff
@@ -16,37 +16,147 @@ def perform
 
         private
 
+        def network_update_plan(network_name, network_subnets, subnet_models, deployment_cloud_config)
+          old_network = deployment_cloud_config['networks'].find { |x| x['name'] == network_name }
+          old_subnets = old_network['subnets']
```

Comment:
We're having a hard time understanding why there are three sets of subnets to consider. Our current understanding is that `subnet_models` are what are currently saved in the database, and therefore existing subnets, and `network_subnets` are the desired subnets. What do `old_subnets` mean? If there are the subnets that are defined in the cloud config, why are they "old"?